src/psi/j/executors/local.py

```diff
@@ -205,14 +205,17 @@ def submit(self, job: Job) -> None:
             raise InvalidJobException('Missing specification')
 
         launcher = self._get_launcher(self._get_launcher_name(spec))
-        args = launcher.get_launch_command(spec)
+        args = launcher.get_launch_command(job)
 
         p = _ChildProcessEntry(job, self)
 
         try:
             with job._status_cv:
                 if job.status.state == JobState.CANCELED:
                     raise SubmitException('Job canceled')
+            if logger.isEnabledFor(logging.DEBUG):
+                logger.debug('Running {},  out={}, err={}'.format(args, spec.stdout_path,
+                                                                  spec.stderr_path))
```

Comment:
The test for `logger.isEnabledFor('DEBUG')` is already performed by the implementation of `logger.debug` - so why are the logging calls shielded?  I am probably missing something...

If it is to avoid early string expansion on the `format`, that is usually addressed by:
```py
logger.debug('Running %s,  out=%s, err=%s', args, spec.stdout_path, spec.stderr_path)
```
the logging module will the only expand if the respective log level is enabled.  There is probably an option for `format` like string expansion as you seem to prefer that, but I don't actually know...

Reply:
Yes, but the check in logger.debug comes after the message string is interpolated.

Reply:
That said, I should probably use the built-in formatting, as you suggest.

Reply:
Removed the guards in the local executor. Will also remove in other places, but not in this PR.